SMLHelper/Options/ModOptions.cs

```diff
@@ -14,7 +15,7 @@ public abstract partial class ModOptions
         /// <summary>
         /// The name of this set of configuration options.
         /// </summary>
-        public readonly string Name;
+        public string Name { get; protected set; }
```

Comment:
Same as `AddSliderOption` here, it'll break any mod that uses this field. Though I don't think that anyone is using it outside of SMLHelper, so it's probably safe in this case.

Reply:
Yeah, I strongly doubt anyone is using this field outside of SMLHelper and if they are... probably a badly written mod in the first place? Not that I'm saying that makes it okay to cause them a breaking change (all they'd need to do is recompile) but... eh...

Reply:
Turns out someone was using it, so we've reverted it back to a `public field` (dropped the `readonly`).

In testing, this is working fine. It _does_ have the side-effect that now people could technically change their menu names on-the-fly - and even I suppose the names of other peoples' menus.

Technically, they could already do that with some reflection, but with this change they don't even need to know how to do that. So I'm a little worried that suddenly modders will be tempted to screw with each others' menu names... but probably not?

For anyone reading this comment and wondering what the purpose of this change is:
This PR is aiming to allow you to use Attributes to set the data needed for creating your menu. In the past, the only way to set the name of your menu was by passing it as a parameter to the base `ModOptions` constructor:

```cs
public class Options : ModOptions
{
    public Options() : base("My options menu") { }
}
```
And the `Name` field is then set in the `ModOptions` constructor from this parameter.

Since the `Name` field is `readonly`, this means that _only_ the original `ModOptions` constructor can set it - no derivatives can, not even in their constructors. And since we now want to set the `Name` from a derivative after parsing a `MenuAttribute` for its `Name` property, this means we're stuck unless we use reflection to set it (inefficient, feels dirty, not supported in .NET Core) or drop the `readonly` attribute from the field.

It's rather a shame it was chosen to be a `public readonly string` field. A `public` getter with a `protected` setter would have made for a much easier to extend API.